src/Speckle.Sdk/Serialisation/Send/SendProcess.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Threading.Channels;
+using Open.ChannelExtensions;
+using Speckle.Sdk.Common;
+using Speckle.Sdk.Models;
+using Speckle.Sdk.Transports;
+using Speckle.Sdk.Transports.ServerUtils;
+
+namespace Speckle.Sdk.Serialisation.Send;
+
+public record SendProcessSettings(
+  int MaxSerializeThreads = 4,
+  int MaxObjectRequestSize = ServerApi.BATCH_SIZE_GET_OBJECTS,
+  int BatchWaitMilliseconds = 500
+);
+
+public sealed class SendProcess : IDisposable
+{
+  private readonly SendProcessSettings _settings = new();
+
+  public SendProcess(IModelTarget modelTarget)
+  {
+    SourceChannel = Channel.CreateUnbounded<Base>();
+    SendStage = new(modelTarget);
+    SerializeStage = new();
+  }
+
+  private long _requested;
+  private Serialized? _rootObjectSerialized;
+  private Base? _rootObject;
+  public Channel<Base> SourceChannel { get; }
+  public SerializeStage SerializeStage { get; set; }
+  public SendStage SendStage { get; set; }
+
+  public Action<ProgressArgs[]>? Progress { get; set; }
+
+  public void InvokeProgress() =>
```

Comment:
Is progress reporting not implemented? or is this simply setting a "completed" progress when an operation is complete?

Reply:
Nope. this is doing progress is reported here, just total is null due to limitaitons